### PR TITLE
Sort by any field in cms

### LINKED ISSUE #2774: Sort posts by date in CMS
## Summary

As a content editor, I want to edit the **most recent post** so that I can fix a typo that just went out. 

The problem is that I don't know the filename, and can't easily find this post. I can't sort by date, nor can I search the filename (related to [this issue](https://github.com/tinacms/tinacms/issues/2723))

Screen Shot 2022-04-05 at 11 33 56 AM

### Expected Results

* See post title and meta information in list (date last edited, author, etc)
* Sort by fields

### PR DESCRIPTION
**Pull request by logan-anderson**
This adds the ability to sort in the CMS

closes #2774

### CODE REVIEW

**Comment by jamespohalloran** [comment]
Sweet!! This one's going to be so helpful.

Small thing @spbyrne (and don't think it needs to hold this PR up) but the loading state flash is a bit jarring:
https://www.loom.com/share/72947a36c6d44874a04f899250113369

File: packages/@tinacms/graphql/src/database/index.ts [hunk 1]
```diff
@@ -483,7 +483,10 @@ export class Database {
             cursor: btoa(edge.cursor),
           }
         } catch (error) {
-          if (error instanceof Error) {
+          if (
+            error instanceof Error &&
+            !edge.path.includes('.tina/__generated__/_graphql.json')
```
Comment (kldavis4) [inline packages/@tinacms/graphql/src/database/index.ts:488]: @logan-anderson is this change related to the sorting feature?

File: packages/tinacms/src/admin/pages/CollectionListPage.tsx [hunk 2]
```diff
@@ -174,11 +208,70 @@ const CollectionListPage = () => {
...
+                                    label: 'No Selection',
+                                    value: JSON.stringify({
+                                      order: 'asc',
+                                      name: '',
+                                    }),
+                                  },
+                                  ...fields
+                                    .map((x) => [
+                                      {
+                                        label: x.label + ' (Acscending)',
```
Comment (kldavis4) [inline packages/tinacms/src/admin/pages/CollectionListPage.tsx:239]:
```suggestion
                                        label: x.label + ' (Ascending)',
```

**Comment by kldavis4** [comment]
Instead of "No Selection" should it say Default maybe (which is the filename sort)?

**Comment by jamespohalloran** [comment]
> Instead of "No Selection" should it say Default maybe (which is the filename sort)?

I may be wrong on this, but I thought someone said recently the default sort was the isTitle field (if it's set)

**Comment by kldavis4** [comment]
> > Instead of "No Selection" should it say Default maybe (which is the filename sort)?
> 
> I may be wrong on this, but I thought someone said recently the default sort was the isTitle field (if it's set)

yeah, I think if isTitle is set then that field is the default, otherwise it is the filename. Either way, a "default" sort is applied - "No selection" seems to be saying that there is no sort.

File: packages/@tinacms/graphql/src/database/index.ts (hunk 1 above)
Comment (logan-anderson) [reply packages/@tinacms/graphql/src/database/index.ts:488]: Sorta. I was getting a strange error message when an invalid sort key was given. This changes to just the raw error message

**Comment by logan-anderson** [comment]
> Sweet!! This one's going to be so helpful.
> 
> Small thing @spbyrne (and don't think it needs to hold this PR up) but the loading state flash is a bit jarring: https://www.loom.com/share/72947a36c6d44874a04f899250113369

@jamespohalloran Do you have a suggestion of how to fix this? In production the loading will take a bit longer and the loading message will be very helpful and feel better.

**Comment by logan-anderson** [comment]
> > > Instead of "No Selection" should it say Default maybe (which is the filename sort)?
> > 
> > 
> > I may be wrong on this, but I thought someone said recently the default sort was the isTitle field (if it's set)
> 
> yeah, I think if isTitle is set then that field is the default, otherwise it is the filename. Either way, a "default" sort is applied - "No selection" seems to be saying that there is no sort.

@jamespohalloran @kldavis4 

The default is the 'IsTitle" field if there is no isTitle field then it will be the file name. 

I can change is to "Default" instead of the "No selection"

**Review by kldavis4 — APPROVED** [review]

**Comment by jamespohalloran** [comment]
> 

@logan-anderson Here's how Linear does it: https://www.loom.com/share/021159a491b744ca8f6a9fb0cd608cdc
It just looks a bit smoother without the global loading state flicker.
